features/messages/impl/src/test/kotlin/io/element/android/features/messages/impl/MessagesPresenterTest.kt

```diff
@@ -126,6 +126,20 @@ class MessagesPresenterTest {
         }
     }
 
+    @Test
+    fun `present - call is disabled if user cannot join it`() = runTest {
+        val room = FakeMatrixRoom().apply {
+            givenCanUserJoinCall(Result.success(false))
+        }
+        val presenter = createMessagesPresenter(matrixRoom = room)
+        moleculeFlow(RecompositionMode.Immediate) {
+            presenter.present()
+        }.test {
+            val initialState = consumeItemsUntilTimeout().last()
+            assertThat(initialState.callState).isEqualTo(RoomCallState.DISABLED)
```

Comment:
Ideally I think we should test that the value is still `DISABLED` when there is an ongoing call. (the test may be added later I guess)

Reply:
Just for my understanding, the user cannot create a call if the power level is too low, this is OK, but they cannot join an ongoing call neither? That looks weird, but I may misreading https://github.com/vector-im/element-meta/issues/2229. I do not know when `m.call.member` are sent, but probably when joining the call. Technically OK, but for the user POV it's a bit strange.

Reply:
AFAIK the `m.call.member` event is sent when the user joins (or starts, if it's the 1st user) a call.

Reply:
OK, thanks

Reply:
This is correct.
The logic indeed is: no permission for m.call.member events -> no calls. (There are discussions about making this the default behaviour or changing the spec so that it is allowed by default. But for now rooms that dont make it possibel to send the `m.call.member` event just dont provide the feature to do calls)